### PR TITLE
Feature/isloading

### PR DESCRIPTION
**Pull request by aaronmaturen**

### CODE REVIEW

File: website/react-magma-docs/src/pages/api/button.mdx [hunk 1]
```diff
@@ -140,6 +140,24 @@ export function Example() {
 }
 ```
 
+## Loading
+
+When a button is in a loading state, the label will be replaced with a loading spinner.
```
Comment (joasmi723) [inline website/react-magma-docs/src/pages/api/button.mdx:145]:
```suggestion
When a button is in a loading state, the label can be replaced with a loading spinner using the `isLoading` prop.
```

Comment (joasmi723) [reply website/react-magma-docs/src/pages/api/button.mdx:145]: Can you add a similar example to the IconButton mdx page as well?

File: packages/react-magma-dom/src/components/StyledButton/index.tsx [hunk 2]
```diff
@@ -17,11 +17,13 @@ import {
...
 import { ThemeContext } from '../../theme/ThemeContext';
-import { ButtonProps, ButtonType } from '../Button';
+import { ButtonType, ButtonProps, ButtonSize, ButtonVariant } from '../Button';
+import { Spinner } from '../Spinner';
 
-interface StyledButtonProps extends ButtonProps {
+export interface StyledButtonProps extends ButtonProps {
   href?: string;
   iconOnly?: boolean;
+  isLoading?: boolean;
```
Comment (joasmi723) [inline packages/react-magma-dom/src/components/StyledButton/index.tsx:26]: I _think_ the `isLoading` prop should be specified in the `BaseButtonProps` or `ButtonStyles` interface instead and extended from here. 

This seems to be working though (but I'm not sure why), except for the fact that the prop isn't showing up in the props table.

Comment (joasmi723) [reply packages/react-magma-dom/src/components/StyledButton/index.tsx:26]: Yes, I double checked.  This prop should be removed here and added to the `ButtonStyles` interface in Button/index.tsx.  When you move it there, make sure you add a comment above it that describes the prop for the props table.  For example:

```
/**
  *  Set the button to a loading state
  * @default false
  */
  isLoading?: boolean;
```

Thanks!

Comment (jamiesmyln) [reply packages/react-magma-dom/src/components/StyledButton/index.tsx]: Ok, it is now in Button/index.tsx.

File: packages/react-magma-dom/src/components/Button/index.tsx [hunk 3]
```diff
@@ -56,6 +56,11 @@ export interface ButtonStyles {
    * Defines the border radius
    * @default ButtonShape.fill
    */
+  /**
```
Comment (joasmi723) [inline packages/react-magma-dom/src/components/Button/index.tsx:59]: Looks like you broke up the description of the `shape` prop with the prop itself.

**Review by joasmi723 — APPROVED** [review]

**Pull request merged**